designsafe/apps/auth/views.py

```diff
@@ -107,8 +110,10 @@ def agave_oauth_callback(request):
     """
     http://agaveapi.co/documentation/authorization-guide/#authorization_code_flow
     """
+    load_alerts_into_session(request)
```

Comment:
I think this would probably need to be in a middleware. If somebody creates a new alert when a user is already logged in that user is not going to see this.
If we put it in a middleware then we can add the alerts to the session every time a user does a request to the user.
Then we find another issue, if that's a request within the data depo the user is not going to see it unless they fully refresh the page.
We could take care of this but I think that's out of this scope. For now we can move this to a middle ware and then create another ticket to see how can we show alerts within a section in the portal that only does partial updates.

Reply:
The instructions were to only pull alerts on login. I brought up the same issues you brought up with this and learned that is expected behavior.